src/core/mesh/qgsmeshvectorrenderer.cpp

```diff
@@ -542,6 +542,8 @@ QgsMeshVectorWindBarbRenderer::QgsMeshVectorWindBarbRenderer(
         context,
         size )
 {
+  const QgsCoordinateReferenceSystem mapCrs = mContext.coordinateTransform().destinationCrs();
+  mMapToWgs = QgsCoordinateTransform( mapCrs, QgsCoordinateReferenceSystem::fromEpsgId( 4326 ), mContext.coordinateTransform().context() );
```

Comment:
fromEpgsId should probably be deprecated now, let's go with:

```suggestion
  mMapToWgs = QgsCoordinateTransform( mapCrs, QgsCoordinateReferenceSystem( QStringLiteral("EPSG:4326" ), mContext.coordinateTransform().context() );
```

BUT -- how will this work with non-earth coordinate systems? I'd check the map's crs celestialBodyName before attempting this logic.

Reply:
> fromEpgsId should probably be deprecated now

Ah, I had a deja-vu about using this, I had to check the headers to verify it's not deprecated! Let's deprecate it then!





> BUT -- how will this work with non-earth coordinate systems?

If the transform does not exist or the transformation fails we default to the _North hemisphere_ mode.

Extra-terrestrial meteorologists working on their celestial body's southern hemisphere will be disappointed though...